_posts/2018-02-26-success-with-serverless.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+status: publish
+published: true
+title: 'Success With Serverless'
+author:
+  display_name: Scott Davis
+  email: stdavis@utah.gov
+date: 2018-02-26 23:45:28
+categories:
+- Developer
+tags:
+- nodejs
+---
+
+In this post, I'd like to describe a recent success story that we've had using [serverless computing](https://en.wikipedia.org/wiki/Serverless_computing). This new(ish) technology has some exciting advantages over traditional solutions. We were looking for an excuse to give it a try and found a great fit.
+
+### The Problem
+[AGRC's base maps]({{ "/data/base-map-and-imagery/" | prepend: site.baseurl }}) (including the [Google imagery]({{ "/discover/license/" | prepend: site.baseurl }})) are served via a custom server application called Giza. Part of the advantage of this solution is the ability to secure and track usage via quad-words. These unique words are assigned to a specific user and are locked down to a specific domain or IP address. For example, if my quad-word is locked down to <code>mydomain.com</code>, then requests originating from any other domain or IP address are blocked by the server. This prevents unauthorized access of licensed content as well as allows AGRC to track analytics.
+
+This quad-word system works great until you try to use one of Esri's out-of-the-box print services ([here's an example of one](http://mapserv.utah.gov/arcgis/rest/services/Utilities/PrintingTools/GPServer)). When you send a web map to this service, it reconstructs all of the layers on the server. This causes requests for base maps to be sent from your ArcGIS Server box, rather than your user's browser (with your domain as the referrer). We do allow wide open quad-words (not locked down to any domain/IP) to be used by those who need to make requests from servers or other local machines. However, these wide open quad-words cannot be used in web applications due to the fact that they could be copied and used by unauthorized users.
```

Comment:
_"This quad-word system works great until you try to use one of Esri’s out-of-the-box print services (here’s an example of one)."_: Would "This quad-word system works great . . . until you try to use one of Esri’s out-of-the-box print services (here’s an example of one)." convey the right tone here? (i.e., adding the ellipsis after "great")?
_"When you send a web map to this service,"_: What service? Not completely clear to me. 
_"We do allow wide open quad-words"_: Can we hyphenate "wide-open" or is that a technical term? If we do hyphenate the term, we'll need to make sure it's hyphenated consistently. 
_"(not locked down to any domain/IP)"_: Should we add "address" after "IP"?

Reply:
"..." is fine with me.

"service" is referring to the print service in the preceding sentence.

Hyphenating "wide-open" is fine with me. It's just a term that we made up.

Reply:
Ok, got it! I think the "services" (plural) and "service" (singular) threw me off a little bit. Maybe we could do: "This quad-word system works great. . . until you try to use one of Esri’s out-of-the-box print services (here’s an example of one). When you send a web map to one of these services, the service reconstructs all of the layers on the server."